### PR TITLE
Medical database

### PR DESCRIPTION
**Pull request by omar-muhamad**
In this project we did:

- [x] [Add patients, medical_histories, and treatments tables](https://github.com/Omar-Muhamad/Medical-Database/commit/3e4595303e7466e71a773100c51eae1482e70367).
- [x] [Add tables invoices, invoice_items, join tables treatment_histories](https://github.com/Omar-Muhamad/Medical-Database/commit/bcff3d9e40632bd79c912a4a9da96a0ae6034372).
- [x] [Edit readme file](https://github.com/Omar-Muhamad/Medical-Database/commit/38dcaa75484cb78f67230f2c44e668238a1b5464).

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  id INT GENERATED ALWAYS AS IDENTITY,
+  unit_price DECIMAL,
+  quantity INT,
+  total_price DECIMAL,
+  invoice_id INT,
+  treatment_id INT,
+  PRIMARY KEY (id),
+  FOREIGN KEY (invoice_id) REFERENCES invoices(id),
+  FOREIGN KEY (treatment_id) REFERENCES treatments(id)
+);
```
Comment (julie-ify) [inline schema_based_on_diagram.sql:54]: - [x] Let's make sure that all the `FOREIGN KEYS` have indexes created. For example:
```ruby
  CREATE INDEX ON invoice_items (treatment_id);
  CREATE INDEX ON invoice_items (invoice_id);

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+DROP TABLE IF EXISTS treatment_histories;
+CREATE TABLE treatment_histories (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  medical_history_id INT,
+  treatment_id INT,
+  PRIMARY KEY (id),
+  FOREIGN KEY (medical_history_id) REFERENCES medical_histories(id),
+  FOREIGN KEY (treatment_id) REFERENCES treatments(id)
```
Comment (julie-ify) [inline schema_based_on_diagram.sql:66]: - [x] Let's also make sure that all the `FOREIGN KEYS` have indexes created.

File: schema_based_on_diagram.sql [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+DROP TABLE IF EXISTS medical_histories;
+CREATE TABLE medical_histories (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  admitted_at TIMESTAMP,
+  patient_id INT,
+  status VARCHAR(255),
+  PRIMARY KEY (id),
+  FOREIGN KEY (patient_id) REFERENCES patients(id)
+);
```
Comment (julie-ify) [inline schema_based_on_diagram.sql:18]: - [x] Let's also make sure that the `FOREIGN KEY` has an index created.

File: schema_based_on_diagram.sql [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+DROP TABLE IF EXISTS invoices;
+CREATE TABLE invoices (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  total_amount DECIMAL,
+  generated_at TIMESTAMP,
+  payed_at TIMESTAMP,
+  medical_histories_id INT,
+  PRIMARY KEY (id),
+  FOREIGN KEY (medical_histories_id) REFERENCES medical_histories(id)
+);
```
Comment (julie-ify) [inline schema_based_on_diagram.sql]: - [x] Let's ensure that we follow the ERD by making sure that the names of the column in the diagram provided are the same as yours. Please consider changing this `medical_histories_id` column to `medical_history_id` so as to match the ERD provided 👍.

- [x] Let's also make sure that the `FOREIGN KEY` has an index created.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Omar-Muhamad, and @qwibbler,

Good job so far!

## Highlight!
- Good feature branch name ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ZeenatLawal — APPROVED** [review]
Hi @Omar-Muhamad and @qwibbler, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

## Congratulations! 🎉 
6vEo


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**